mobile_app_connector/models/compassion_correspondence.py

```diff
@@ -221,9 +221,16 @@ def mobile_send_letter(self, *params, **parameters):
             )
         )
         # Use the latest preview for generating the letter
-        gen[:1].generate_letters()
+        # update utms
+        utm_source = params.get("utm_source")
+        utm_medium = params.get("utm_medium")
+        utm_campaign = params.get("utm_campaign")
+        utms = self.env["utm.mixin"].get_utms(utm_source, utm_medium, utm_campaign)
```

Comment:
The method `get_utms` is defined in module `mass_mailing_switzerland` and is therefore not available from here. I think you should change and assume that the utms are already transmitted as a dictionary (call the `get_utms` method before calling the generator).
```suggestion
        utms = params.get("utms")
```

Reply:
Please also update the comment at line 184 to mention that the params can contain this `utms` value and how it should be passed in order to work.